src/MongoDB.Driver/PipelineDefinition.cs

```diff
@@ -352,8 +352,8 @@ private static List<IPipelineStageDefinition> VerifyStages(List<IPipelineStageDe
             {
                 var message = string.Format(
                     "The output type to the last stage was expected to be {0}, but was {1}.",
-                    nextInputType,
-                    stages.Last().OutputType);
+                    typeof(TOutput),
+                    nextInputType);
```

Comment:
I'd rather see you swap `{0}` and `{1}` in the format string to stay consistent with the earlier error message on line 341.